bouncer/tests/broker_fee_collection_test.ts

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env -S pnpm tsx
+import assert from 'assert';
+import { randomBytes } from 'crypto';
+import { KeyringPair } from '@polkadot/keyring/types';
+import Keyring from '@polkadot/keyring';
+import { Asset, Assets, assetDecimals } from '@chainflip-io/cli';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import {
+  EgressId,
+  amountToFineAmount,
+  chainShortNameFromAsset,
+  decodeDotAddressForContract,
+  getChainflipApi,
+  newAddress,
+  observeBalanceIncrease,
+  observeEvent,
+  runWithTimeout,
+  sleep,
+} from '../shared/utils';
+import { getBalance } from '../shared/get_balance';
+import { performSwap } from '../shared/perform_swap';
+
+const testSwapAssetAmount = {
+  [Assets.ETH]: 1,
+  [Assets.DOT]: 1000,
+  [Assets.FLIP]: 1000,
+  [Assets.BTC]: 0.1,
+  [Assets.USDC]: 1000,
+};
+const testCommissionBps = 1000; // 10%
+
+// Maximum expected deposit and withdrawal fees
+const maxDepositFee = {
+  [Assets.ETH]: BigInt(350000),
+  [Assets.DOT]: BigInt(197300000),
+  [Assets.FLIP]: BigInt(300000000),
+  [Assets.BTC]: BigInt(190),
+  [Assets.USDC]: BigInt(0), // Fee is too low for localnet, it rounds to 0
+};
+const maxWithdrawalFee = {
+  [Assets.ETH]: BigInt(490000),
+  [Assets.DOT]: BigInt(197450000),
+  [Assets.FLIP]: BigInt(300000000),
+  [Assets.BTC]: BigInt(100),
+  [Assets.USDC]: BigInt(0),
+};
+const chainflip = await getChainflipApi();
+
+let withdrawLockout = false;
+let broker1: KeyringPair | undefined;
+export async function broker1KeyringPair() {
+  if (!broker1) {
+    await cryptoWaitReady();
+    const keyring = new Keyring({ type: 'sr25519' });
+    broker1 = keyring.createFromUri('//BROKER_1');
+  }
+  return broker1;
+}
+
+/// Runs a swap, checks that the broker fees are collected,
+/// then withdraws the broker fees, making sure the balance is correct after the withdrawal.
+async function testBrokerFees(asset: Asset, seed?: string): Promise<void> {
+  const broker = await broker1KeyringPair();
+
+  // Check the broker fees before the swap
+  const earnedBrokerFeesBefore = BigInt(
+    (await chainflip.query.swapping.earnedBrokerFees(broker.address, asset)).toString(),
+  );
+  console.log(`${asset} earnedBrokerFeesBefore:`, earnedBrokerFeesBefore);
+
+  // Run a swap
+  const swapAsset = asset === Assets.USDC ? Assets.FLIP : Assets.USDC;
+  const destinationAddress = await newAddress(swapAsset, seed ?? randomBytes(32).toString('hex'));
+  const useableDestinationAddress =
```

Comment:
I'm not really sure what to call it because im not really sure why we have this different type of address. Calling it `observeDestinationAddress` for now. 🤷‍♂️

Reply:
In that case it might make sense to move observeDestinationAddress inside the observeEvent, so its scope is limited. Then maybe you could also call it expectedDestinationAddress.

Reply:
> move observeDestinationAddress inside the observeEvent,

Not quite sure what you mean by this. But the other place that uses the decoded address also uses it for `submitBrokerWithdrawal`. So its not just an observe address really.  

Maybe @martin-chainflip or @kylezs could explain why we need this address and help us clean this up.

Reply:
As discussed, was just referring to that one variable that could be moved inside the closure so that we don't pollute the outer variable scope, but I see that you do something similar further down where this wouldn't work, so I don't mind if you want to leave as is for consistency.